### DISCUSSION TITLE
HEIC support with conversion class ?

### DISCUSSION THREAD

**Discussion by dgillier** [spatie/laravel-medialibrary#2978] [Q&A]
Hello,
More and more pictures are coming from phone cameras these days and iPhone are using HEIC, not supported by this library or any browser. 
So a conversion to jpg (or png) is needed...

I created  a medialibrary conversion class using Imagemagick but I don't want to use it for conversion to another item (i.e thumb,) but to have heic files converted to jpg as original (like spatie medialibrary is doing with tiff files).

Anyway to do that ?

Tks ,Denis

**Comment by freekmurze** [comment]
My suggestion would be to converted HEIC files to JPG before adding it to the media library.

**Comment by dgillier** [comment]
Sure, but any reason why we can add more input formats to this library (i.e tiff is supporeted while not a native format and it's currently being converted) ?
